pkg/settings/options.go

```diff
@@ -26,6 +27,32 @@ type SettingOption struct {
 	validateProtoFn    func(*Values, protoutil.Message) error
 }
 
+// NameStatus indicates the status of a setting name.
+type NameStatus bool
+
+const (
+	// NameActive indicates that the name is currently in use.
+	NameActive NameStatus = true
+	// NameRetired indicates that the name is no longer in use.
+	NameRetired NameStatus = false
+)
+
+// WithName configures the user-visible name of the setting.
+func WithName(name SettingName) SettingOption {
+	return SettingOption{commonOpt: func(c *common) {
+		c.setName(name)
+		registerAlias(c.key, name, NameActive)
+	}}
+}
```

Comment:
It looks like this currently permits more than 1 active name.  If we wanted to prevent that without changing this datastructure, perhaps a test or something that only runs in test builds which explodes if you register two names would be useful,